modules/global/src/com/haulmont/addon/ldap/utils/MatchingRuleUtils.java

```diff
@@ -129,18 +134,22 @@ public boolean validateRuleOrder(Integer order) {
         return true;
     }
 
-    public static List<Role> getRoles(User user) {
+    public List<Role> getRoles(User user) {
         return user.getUserRoles().stream()
+                .peek(ur -> {
```

Comment:
won't it be simpler if we do everything inside a single map() mathod instead of two sequential peek and map?